torch/_export/serialize.py

```diff
@@ -46,30 +69,441 @@ def _reverse_map(d):
+
+
+class Serializer:
+    def __init__(self, op_version: int = 0):
+        self.inputs: List[Argument] = []
+        self.outputs: List[Argument] = []
+        self.nodes: List[Node] = []
+        self.tensor_values: Dict[str, TensorValue] = {}
+        self.sym_int_values: Dict[str, SymInt] = {}
+        self.op_version: int = op_version
+
+    def handle_placeholder(self, node: torch.fx.Node):
+        assert node.op == "placeholder"
+        self.inputs.append(Argument.create(as_tensor=TensorArgument(name=node.name)))
+
+        self.tensor_values[node.name] = TensorValue(
+            meta=export_tensor_meta(node.meta["val"])
+        )
+
+    def handle_output(self, node: torch.fx.Node):
+        assert node.op == "output"
+        assert len(node.args) == 1, "FX.Node's args should have one arg"
+        node_args = node.args[0]
+        assert isinstance(node_args, list)
+        self.outputs = [self.export_input(arg) for arg in node_args]
+
+    def handle_call_function(self, node: torch.fx.Node):
+        assert node.op == "call_function"
+
+        # getitem has been handled in the producer node, skip it here
+        if node.target is operator.getitem:
+            return
+
+        if node.target is torch.set_grad_enabled:
+            # Hack for torch.no_grad support. In the long run this should become
+            # a higher order op but this is fine for now. See [NOTE: nograd support]
+            ex_node = Node(
+                target=export_operator("torch.set_grad_enabled", self.op_version),
+                inputs=[NamedArgument(name="arg", arg=self.export_input(node.args[0]))],
+                outputs=[],
+                metadata=export_metadata(node),
+            )
+        elif node.target in _SYM_INT_OPS:
+            assert len(node.kwargs) == 0
+            meta_val = node.meta["val"]
+            ex_node = Node(
+                target=export_operator(node.target, self.op_version),
+                inputs=self.export_sym_int_op_inputs(node.args),
+                outputs=[Argument.create(as_sym_int=self.export_sym_int_output(node.name, meta_val))],
+                metadata=export_metadata(node),
+            )
+        elif isinstance(node.target, torch._ops.OpOverload):
+            ex_node = Node(
+                target=export_operator(node.target, self.op_version),
+                inputs=self.export_inputs(node.target, node.args, node.kwargs),
+                outputs=self.export_outputs(node),
+                # TODO: create a new tensor_values here, meta might have faketensor info
+                metadata=export_metadata(node),
+            )
+        else:
+            # TODO(angelayi) Higher order ops
+            raise SerializeError(f"Serializing {node.target} is not supported")
+
+        self.nodes.append(ex_node)
+
+    def handle_get_attr(self, node):
+        pass
+
+    def export_sym_int_op_inputs(self, args) -> List[NamedArgument]:
+        serialized_args = []
+        args_names = ["a", "b"]
```

Comment:
What is "a" and "b"?

Reply:
Arbitrary names because those ops don't actually have an arg name :P